docs/Adding-Software-to-Homebrew.md

```diff
@@ -186,6 +237,24 @@ Other commonly used stanzas are:
 
 Additional [`artifact` stanzas](Cask-Cookbook.md#at-least-one-artifact-stanza-is-also-required) may be needed for special use cases. Even more special-use stanzas are listed at [Optional Stanzas](Cask-Cookbook.md#optional-stanzas).
 
+#### Generating a token for the cask
```

Comment:
This section should be moved to the Cask Cookbok IMO; maybe still linked to as a 1-liner here if super relevant.

This is something I never really think about these days as I just use 'brew create'.

Reply:
Agreed, this feels like it could be dropped from the documentation entirely and probably predates `brew create` being as smart as it is (or maybe even supporting casks at all)

Reply:
I'm inclined not to remove it entirely, as there may be cases where this level of knowledge is useful to know; and that feels like it belongs in a section of the Cask Cookbook.

I wish more of the dev tools were explicitly mentioned/documented in the Cookbooks/relevant docs, rather than seemingly only stumbling upon them by accident when someone tells me about it, or I decide to read through the code, etc.

Reply:
> and that feels like it belongs in a section of the Cask Cookbook.

Agreed.

Reply:
> I wish more of the dev tools were explicitly mentioned/documented in the Cookbooks/relevant docs

Although: I'm not sure this one needs to exist at all. I've made a bunch of casks and never once used or known about this tool until today. CC @homebrew/cask folks for thoughts if we need to maintain this code.

Reply:
I think `brew create` gets you to the same place as the script.

Reply:
> I think `brew create` gets you to the same place as the script.

`brew create` seems to use `token = Cask::Utils.token_from(name)`:

- https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/create.rb#L94

Which is defined here:

- https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/utils.rb#L109-L118

Whereas `generate_cask_token` is defined here:

- https://github.com/Homebrew/homebrew-cask/blob/master/developer/bin/generate_cask_token

And has a TODO note about merging into `brew create`:

- https://github.com/Homebrew/homebrew-cask/blob/master/developer/bin/generate_cask_token#L4-L11

```ruby
# generate_cask_token
#
# todo:
#
# detect Cask files which differ only by the placement of hyphens.
#
# merge entirely into "brew create" command
```

From a quick skim, `generate_cask_token` seems more comprehensive currently.

Reply:
I think the documentation still shouldn't reference this any more. After this PR is merged and the reference is removed, an issue can be created to migrate this logic to `brew create`.

Reply:
@MikeMcQuaid If that's opinion only, then I disagree, and feel that the correct time to remove it would be when that functionality has been migrated, as part of that migration.

If that is a "direct order/overruling my judgement" and a blocker on this PR landing.. I'll begrudgingly accept it.

Reply:
> If that is a "direct order/overruling my judgement"

Yes. Maintainers' judgement trumps yours here, please.